src/osd/OSD.cc

```diff
@@ -9354,9 +9354,10 @@ void OSD::ShardedOpWQ::_enqueue_front(pair<spg_t, PGQueueable> item)
   sdata->sdata_op_ordering_lock.Lock();
   auto p = sdata->pg_slots.find(item.first);
   if (p != sdata->pg_slots.end() && !p->second.to_process.empty()) {
-    // we may be racing with _process, which has dequeued an item from
-    // pqueue, put it on to_process, and is now busy taking the pg
-    // lock.  ensure this op is ordered before any such op in to_process.
+    // we may be racing with _process, which has dequeued a new item
+    // from pqueue, put it on to_process, and is now busy taking the
+    // pg lock.  ensure this old equeued item is ordered before any
+    // such newer item in to_process.
```

Comment:
Yeah, I still think this is broken with respect to a to_process list of >1 entry. Need to take a reference prior to pushing front, or traverse it, or something?